addon/v1alpha1/types_clustermanagementaddon.go

```diff
@@ -175,10 +185,146 @@ type PlacementStrategy struct {
+	//   the concurrence rate defined in MaxConcurrency.
+	// - RollingUpdateWithCanary: when configs change, wait and check if add-ons on the canary placement
+	//   selected clusters have applied the new configs and are healthy, then apply the new configs to
+	//   all the selected clusters with the concurrence rate defined in MaxConcurrency.
+	//
+	//   The field lastKnownGoodConfig in the status record the last successfully applied
+	//   spec hash of canary placement. If the config spec hash changes after the canary is passed and
+	//   before the rollout is done, the current rollout will continue, then roll out to the latest change.
+	//
+	//   For example, the addon configs have spec hash A. The canary is passed and the lastKnownGoodConfig
+	//   would be A, and all the selected clusters are rolling out to A.
+	//   Then the config spec hash changes to B. At this time, the clusters will continue rolling out to A.
+	//   When the rollout is done and canary passed B, the lastKnownGoodConfig would be B and
+	//   all the clusters will start rolling out to B.
+	//
+	//   The canary placement does not have to be a subset of the install placement, and it is more like a
+	//   reference for finding and checking canary clusters before upgrading all. To trigger the rollout
+	//   on the canary clusters, you can define another rollout strategy with the type RollingUpdate, or even
+	//   manually upgrade the addons on those clusters.
+	//
+	// +kubebuilder:validation:Enum=UpdateAll;RollingUpdate;RollingUpdateWithCanary
+	// +kubebuilder:default:=UpdateAll
+	// +optional
+	Type string `json:"type"`
+
+	// Rolling update with placement config params. Present only if the type is RollingUpdate.
+	// +optional
+	RollingUpdate *RollingUpdate `json:"rollingUpdate,omitempty"`
+
+	// Rolling update with placement config params. Present only if the type is RollingUpdateWithCanary.
+	// +optional
+	RollingUpdateWithCanary *RollingUpdateWithCanary `json:"rollingUpdateWithCanary,omitempty"`
+}
+
+const (
+	// AddonRolloutStrategyUpdateAll is the addon rollout strategy representing apply the new configs to
+	// all the selected clusters at once.
+	AddonRolloutStrategyUpdateAll string = "UpdateAll"
+	// AddonRolloutStrategyRollingUpdate is the addon rollout strategy representing apply the new configs to
+	// all the selected clusters with the concurrency rate.
+	AddonRolloutStrategyRollingUpdate string = "RollingUpdate"
+	// AddonRolloutStrategyRollingUpdate is the addon rollout strategy representing wait and check
+	// if add-ons on the canary have applied the new configs, then apply the new configs to
+	// all the selected clusters with the concurrency rate.
+	AddonRolloutStrategyRollingUpdateWithCanary string = "RollingUpdateWithCanary"
+)
+
+// RollingUpdate represents the behavior to rolling update add-on configurations
+// on the selected clusters.
+type RollingUpdate struct {
+	// The maximum concurrently updating number of addons.
```

Comment:
maximum concurrently updating number of clusters?  This struct is for a single add-on on multiple managedclusters, correct?

Reply:
yes, it's for a single add-on on multiple managedclusters. Update the comments.